configurable-rules/azure-apim-unsupported-keyword/README.md

```diff
@@ -0,0 +1,193 @@
+
+This ruleset can help validate the output of the decorator is viable for Azure APIM.
+
+Azure APIM isn't prescriptive if unsupported features break the import/add/update process or if they are silently ignored.
+Users have a reported a combination of those behaviors.
+
+## Code
+
+The first rule checks that the `operationId` conforms to the Azure APIM expectations of kebab-case and 76 max characters.
+It is set to be a warning level severity because Azure APIM automatically transforms non-compliant `operationId` values.
+
+```yaml
+rules:
+  rule/operation-pattern-azure-apim:
+    subject: 
+      type: Operation
+      property: operationId
+    assertions: 
+      casing: kebab-case
+      maxLength: 76
+    severity: warn
+```
+
+The second rule checks that the operation `summary` is 300 characters max length.
+```yaml
+  rule/summary-length-azure-apim:
+    subject: 
+      type: Operation
+      property: summary
+    assertions: 
+      maxLength: 300
+    severity: warn
+```
+
+The next rules highlights if there would be any servers that are ignored (and if you have any that don't use HTTPS):
+
+```yaml
+  rule/multiple-servers-ignored-azure-apim:
+    subject: 
+      type: ServerList
+    assertions: 
+      maxLength: 1
+    message: If multiple servers are specified, API Management will use the first HTTPS URL it finds.
+    severity: warn
+  
+  rule/servers-not-https-azure-apim:
+    subject: 
+      type: Server
+      property: url
+    assertions: 
+      pattern: '^https:\/\/.*'
+    message: Server URL must start with HTTPS.
+```
+
+The next rule checks that schema is not defined inline (uses a `$ref` for MediaType `schema`).
+It also checks that the reference object link is to the components section of the same file which means it starts with `#/components`. 
+
+```yaml
+  rule/mediatype-schema-ref-pattern-azure-apim:
+    subject: 
+      type: MediaType
+      property: schema
+    message: Inline schema definitions and refs that point to URLs or files aren't supported.
+    assertions:     
+      ref: /^#\/components\/.*/
+```
+
+The next two rules ensure external docs and info summary aren't used.
+
+```yaml
+  rule/external-docs-unsupported-azure-apim:
+    subject: 
+      type: ExternalDocs
+    assertions: 
+      defined: false
+    message: Azure APIM does not support externalDocs.
+  
+  rule/info-summary-unsupported-azure-apim:
+    subject: 
+      type: Info
+      property: summary
+    assertions: 
+      defined: false
+    message: Azure APIM does not Info summary.
+```
+
+APIM ignores security. This rule let's you know if it is left in there:
+
+```yaml
+  rule/security-ignored-azure-apim:
+    subject: 
+      type: SecurityRequirementList
+    assertions: 
+      defined: false
+    message: Azure APIM ignores security.
```

Comment:
severity warn?

Reply:
It could be. I'm unclear if it is fully ignored or if it causes an error upon import.